docs/proposals/1374-multi-cluster-inference/README.md

```diff
@@ -0,0 +1,622 @@
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // ObservedGeneration reflects the generation observed by the controller.
+    //
+    // +kubebuilder:validation:Optional
+    ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+type ImportedCluster struct {
+    // Name of the exporting cluster (must be unique within the list).
+    //
+    // +kubebuilder:validation:Required
+    Name string `json:"name"`
+
+    // ReachabilityMode provides a hint to InferenceGateways how to program data plane
+    // for this import.
+    //
+    // +kubebuilder:validation:Enum=Endpoints;Parents
+    // +kubebuilder:validation:Required
+    ReachabilityMode ReachabilityMode `json:"reachabilityMode"`
+
+    // EndpointPicker describes how to connect to the EPP in the exporting cluster.
+    // Used when ReachabilityMode=Endpoints.
+    //
+    // +kubebuilder:validation:Optional
+    EndpointPicker EndpointPickerImport `json:"endpointPicker,omitempty"`
+
+    // Parents describes optional "parent" frontends (e.g., Gateways) to route through
+    // when ReachabilityMode=Parents.
+    //
+    // +kubebuilder:validation:Optional
+    Parents []ParentImport `json:"parents,omitempty"`
+
+    // Conditions for cluster-scoped readiness (e.g., per-cluster EPP health).
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type EndpointPickerImport struct {
+    // Name of the EPP (informational).
+    //
+    // +kubebuilder:validation:Required
+    Name string `json:"name"`
+
+    // Selector determines which Pods are members of the exported InferencePool.
+    //
+    // +kubebuilder:validation:Required
+    Selector LabelSelector `json:"selector"`
+
+    // TargetPortNumber is the port the model servers listen on in the exported pool.
+    // Used when ReachabilityMode=Endpoints.
+    //
+    // +kubebuilder:validation:Required
+    TargetPortNumber int32 `json:"targetPortNumber"`
+
+    // Service endpoints (addresses + ports) where the EPP exposes ext-proc.
+    //
+    // +kubebuilder:validation:Required
+    Service []ServiceImport `json:"service"`
+
+    // Health endpoint (optional) of the exported EPP.
+    //
+    // +kubebuilder:validation:Optional
+    Health *HealthEndpoint `json:"health,omitempty"`
+
+    // Metrics endpoint (optional) of the exported EPP.
+    //
+    // +kubebuilder:validation:Optional
+    Metrics *MetricsEndpoint `json:"metrics,omitempty"`
+}
+
+// ParentImport models a remote "parent" (typically a Gateway Service) that fronts
+// the imported InferencePool.
+type ParentImport struct {
+    Name       string `json:"name"`
+    Namespace  string `json:"namespace"`
+    Service    []ServiceImport `json:"service"`
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type ServiceImport struct {
+    // Type mirrors core Service types for clarity.
+    // +kubebuilder:validation:Enum=ClusterIP;NodePort;LoadBalancer;ExternalName
```

Comment:
So, implementations I work on are unlikely to use any of this for status and just ignore it entirely (see lots and lots of past discussion on why MCS is a poor fit for service meshes), but I don't see how anyone can use this except LoadBalancer.

A clusterIP is not usable in another cluster, nor is ExternalName

Reply:
Yeah, this is a good point. In the case of GKE, we're using NEGs (roughly similar to EndpointSlices for GCP) to track the endpoints behind the EPP Service, but don't really care about the Service frontend at all here. 

In the common case where there's an intermediate Gateway involved, what kind of information would be useful here?